### ISSUE TITLE
Vim9 script does not support nested keys on assignment

### ISSUE THREAD

**Issue by itchyny** [vim/vim#7309]
_Instructions: Replace the template text and remove irrelevant text (including this line)_

**Describe the bug**
Vim9 script does not support nested keys on assignment. For example, `x[0][0] = 1`.

**To Reproduce**
Load the following script and `:echo F()` and `:echo G()`.
```
def g:F()
  var x: list<number> = [0]
  x[0][0] = 1
  echom x
enddef

def g:G()
  var x: list<list<number>> = [[0]]
  x[0][0] = 1
  echom x
enddef
```

**Expected behavior**
In `F()`, `x[0][0] = 1` should be rejected, but currently it updates the element and results in `[1]`.
In `G()`, `x[0][0] = 1` should be accepted and results in `[[1]]`, but currently it emits `E1012: Type mismatch; expected list<number> but got number`.

**Screenshots**
If applicable, copy/paste the text or add screenshots to help explain your problem.

**Environment (please complete the following information):**
 - Vim version: 8.2.1992
 - OS: macOS 10.15.7
 - Terminal: iTerm2

**Additional context**

**Comment by bfrg** [comment]
This is not implemented yet (see `:h todo`):
https://github.com/vim/vim/blob/4792a679f9e08fc6026a596be3d364cecb70b049/runtime/doc/todo.txt#L63

**Comment by itchyny** [comment]
Okay, thanks. So should we add a new instruction which has arbitrary number of indices, instead of ISN_STORELIST/ISN_STOREDICT?
```
x[idx0][idx1][idx2] = rval
x.key0.key1.key2 = rval
x[idx0].key0[idx1].key1 = rval
x[any * expression][or + something + else] = rval
```
```
PUSH rval
PUSH index0
PUSH index1
...
PUSH indexN
LOAD x
ASSIGN (isn_arg.number = N)
```

**Comment by brammool** [comment]
> Okay, thanks. So should we add a new instruction which has arbitrary number of indices, instead of ISN_STORELIST/ISN_STOREDICT?
> ```
> x[idx0][idx1][idx2] = rval
> x.key0.key1.key2 = rval
> x[idx0].key0[idx1].key1 = rval
> x[any * expression][or + something + else] = rval
> ```
> ```
> PUSH rval
> PUSH index0
> PUSH index1
> ...
> PUSH indexN
> LOAD x
> ASSIGN (isn_arg.number = N)
> ```

This is one part that I haven't worked on yet.  We need something
similar to get_lval(), which figures out the left side of the
assignment, and set_var_lval(), which does the assignment.
Some part has to be done at runtime, and an assignment with += makes it
a bit more complicated.


-- 
hundred-and-one symptoms of being an internet addict:
259. When you enter your name in the AltaVista search engine, the top ten
     matches do indeed refer to you.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by itchyny** [comment]
Closing as already listed in todo.

**Comment by brammool** [comment]
Implemented in patch 8.2.2063
